Add argument to toggle caching of catkin_pkg.packages.find_packages

garaemon opened the pull request:
patch for https://github.com/ros/ros_comm/issues/541

garaemon committed:
Add argument to toggle caching of catkin_pkg.packages.find_packages

dirk-thomas commented:
Please consider to update the PR to use the existing argument `source_path_to_packages` of the `catkin_find` function (see https://github.com/ros/catkin/pull/704#issuecomment-68476827).

garaemon committed:
use source_path_to_packages keyword argument of
catkin.find_in_workspaces.find_in_workspaces and add utility function to
create cache for source_path_to_packages keyword argument

dirk-thomas commented on core/roslib/src/roslib/packages.py:
```diff
@@ -509,3 +515,16 @@ def find_resource(pkg, resource_name, filter_fn=None, rospack=None):
+        source_spaces = catkin_get_source_paths(workspace)
+        for source_path in source_spaces:
+            source_path_to_packages[source_path] = catkin_find_packages(source_path)
+    return source_path_to_packages
```
The logic to iterate over the workspaces and finding packages in each of them is provided by catkin.find_in_workspaces and should be reused instead of being duplicated here.

dirk-thomas commented:
If you are still interested in this feature it would be highly appreciated if you could update the PR based on the comments above.

wkentaro commented:
@dirk-thomas 
#91 is another implementation for this. Please review it.

> The logic to iterate over the workspaces and finding packages in each of them is provided by catkin.find_in_workspaces and should be reused instead of being duplicated here.

Actually, I don't find the way to reuse function in catkin.find_in_workspaces so created another function.
https://github.com/ros/catkin/pull/757
